src/cogent3/core/new_alignment.py

```diff
@@ -2374,3 +2379,559 @@ def _(
+
+
+@seq_to_gap_coords.register
+def _(seq: bytes, moltype: new_moltype.MolType) -> tuple[bytes, IndelMap]:
+    seq, indel_map = seq_to_gap_coords(seq.decode("utf-8"), moltype)
+    return seq.encode("utf-8"), indel_map
+
+
+class SliceRecord(new_sequence.SliceRecordABC):
+    __slots__ = "_seq_len"
+
+    def __init__(
+        self, start: OptInt, stop: OptInt, step: OptInt, offset: int, seq_len: int
+    ):
+        self.start = start
+        self.stop = stop
+        self.step = step
+        self._offset = offset
+        self._seq_len = seq_len
+
+    @property
+    def seq_len(self) -> int:
+        return self._seq_len
+
+    def _get_init_kwargs(self) -> dict:
+        return {}
+
+    def copy(self):
+        # todo: kath
+        ...
+
+    @property
+    def _zero_slice(self): ...
+
+
+class Aligned:
+    """A single sequence in an alignment."""
+
+    def __init__(self, data: AlignedDataView, moltype: MolTypes):
+        self._data = data
+        self._moltype = new_moltype.get_moltype(moltype)
+
+    @property
+    def moltype(self) -> MolTypes:
+        return self._moltype
+
+    @property
+    def data(self) -> AlignedDataView:
+        return self._data
+
+    @property
+    def map(self) -> IndelMap:
+        return self.data.map
+
+    @property
+    def seq(self) -> new_sequence.Sequence:
+        """Returns Sequence object, excluding gaps."""
+        return self.moltype.make_seq(seq=self.data.str_value, name=self.data.seqid)
+
+    def get_gapped_seq(self):
+        """Returns Sequence object, including gaps."""
+        # refactor: design
+        # should we mirror the above property and call this gapped_seq
+        # and make it a property?
+        return self.moltype.make_seq(
+            seq=self.data.gapped_str_value, name=self.data.seqid
+        )
+
+    def __str__(self) -> str:
+        return self.data.gapped_str_value
+
+    def __array__(self) -> numpy.ndarray:
+        return self.data.gapped_array_value
+
+    def __bytes__(self) -> bytes:
+        return self.data.gapped_bytes_value
+
+    def __iter__(self):
+        """Iterates over sequence one motif (e.g. char) at a time, incl. gaps"""
+        yield from self.data.gapped_str_value
+
+    @singledispatchmethod
+    def __getitem__(self, span: Union[int, slice]):
+        raise NotImplementedError(f"__getitem__ not implemented for {type(span)}")
+
+    @__getitem__.register
+    def _(self, span: int):
+        return self.__getitem__(slice(span, span + 1))
+
+    @__getitem__.register
+    def _(self, span: slice):
+        # refactor: design
+        # this is not yet implemented for reverse slices - do we want to do that?
+        # although maybe AlignedDataView should handle this?
+        if span.step and span.step < 0:
+            raise NotImplementedError("Cannot reverse an Aligned")
+
+        return self.__class__(
+            data=self.data[span.start : span.stop], moltype=self.moltype
+        )
+
+
+@dataclass
+class AlignedData:
```

Comment:
Change the name to `AlignedSeqsData` and inherit from `SeqsDataABC`. (We also need to define an `AlignedSeqsDataABC`, which has only methods unique to this class.)

The `AlignedSeqsData` class should mirror the organisation of the `SeqsData` class. It should not be a `dataclass` (only because the use of slots with data classes is not supported at the minimal Python version that Cogent3 currently supports.), but a handcrafted class which has slots defined that are the same as those for `SeqsData`, with the addition of slots for gaps and anything else alignment specific